qualtran/cirq_interop/_bloq_to_cirq.py

```diff
@@ -84,38 +85,10 @@ def bloq_on(
             op: A cirq operation whose gate is the `BloqAsCirqGate`-wrapped version of `bloq`.
             cirq_quregs: The output cirq qubit registers.
         """
-        return _construct_op_from_gate(
-            BloqAsCirqGate(bloq=bloq), in_quregs=cirq_quregs, qubit_manager=qubit_manager
-        )
+        return BloqAsCirqGate(bloq=bloq).as_cirq_op(qubit_manager=qubit_manager, **cirq_quregs)
 
-    def decompose_from_registers(
-        self, context: cirq.DecompositionContext, **quregs: CirqQuregT
-    ) -> cirq.OP_TREE:
-        """Implementation of the GatesWithRegisters decompose method.
-
-        This delegates to `self.bloq.decompose_bloq()` and converts the result to a cirq circuit.
-
-        Args:
-            context: `cirq.DecompositionContext` stores options for decomposing gates (eg:
-                cirq.QubitManager).
-            **quregs: Appropriately shaped qubit arrays corresponding to Cirq-FT registers defined
-                as per `self.signature`.
-
-        Returns:
-            A cirq circuit containing the cirq-exported version of the bloq decomposition.
-        """
-        cbloq = self._bloq.decompose_bloq()
-        circuit, out_quregs = cbloq.to_cirq_circuit(qubit_manager=context.qubit_manager, **quregs)
-        qubit_map = {q: q for q in circuit.all_qubits()}
-        for reg in self.bloq.signature.rights():
-            if reg.side == Side.RIGHT:
-                # Right only registers can get mapped to newly allocated output qubits in `out_regs`.
-                # Map them back to the original system qubits and deallocate newly allocated qubits.
-                assert reg.name in quregs and reg.name in out_quregs
-                assert quregs[reg.name].shape == out_quregs[reg.name].shape
-                context.qubit_manager.qfree([q for q in out_quregs[reg.name].flatten()])
-                qubit_map |= zip(out_quregs[reg.name].flatten(), quregs[reg.name].flatten())
-        return circuit.unfreeze(copy=False).transform_qubits(qubit_map)
+    def decompose_bloq(self) -> 'CompositeBloq':
+        return self.bloq.decompose_bloq()
```

Comment:
a little docstring would not be out of place here

Reply:
done